:sparkles: Update task.svelte to honour theme styling

sigrunixia opened the pull request:
This is a minor tweak that should fix the link styling with certain themes. I added minor font weight increases as well for readability in the cases where the accent-text is more muted.

This should close #12.

sigrunixia commented:
**Note**: I am having a hard time getting a dev environment working with the current setup. I had to update esbuild and esbuild sass dependencies in package.json on my end to get it going. However, it seems like esbuild does not like much of what is happening in Svelte and complains of depreciated features. This goes past my knowledge of "correcting it" however. A check of what dependencies are needed may be needed, and I can open an issue for that if you wish.

sigrunixia commented:
Also, I _dont think_ styles.scss needs updated with this since this is attempting to make these changes baseline. If it does, however, I can add that to this PR.

ivan-lednev commented:
Hello, @sigrunixia! 

Thank you for taking the time to contribute! It's really inspiring when other people want to collaborate with you on your projects ❤️ 

> This is a minor tweak that should fix the link styling with certain themes. 
Unknowingly, I added a different fix for the issue with colors. Check out 0.3.0 and tell me what you think.

I'll investigate the environment issues, and write some documentation in the nearest days, so that it'll be easier to contribute to the project.

ivan-lednev commented on src/ui/components/task.svelte:
```diff
@@ -123,6 +123,18 @@
     transition: 0.05s linear;
   }
 
+  .task a {
```
This is not going to work, since Svelte adds unique classes to every class you use in `.svelte` files. If you want to add some styling to markdown that's rendered inside `task`, use `:global()`, like so:
```css
  .rendered-markdown :global(p) {
    margin-block-start: 0;
    margin-block-end: 0;
  }
```

ivan-lednev commented on src/ui/components/task.svelte:
```diff
@@ -123,6 +123,18 @@
+  .task a:hover {
+    color: var(--text-accent-hover);
+    font-weight: 600;
+    cursor: grab;
```
`grab` is associated with drag-and-drop behavior, so I don't think it makes sense to add it to links. Besides, I try to match Obsidian's styling as much as possible, and Obsidian uses `cursor: pointer` for links.

sigrunixia commented:
> Hello, @sigrunixia!
> 
> Thank you for taking the time to contribute! It's really inspiring when other people want to collaborate with you on your projects ❤️
> 
> > This is a minor tweak that should fix the link styling with certain themes.
> 
> Unknowingly, I added a different fix for the issue with colors. Check out 0.3.0 and tell me what you think.
> 
> I've fixed the installation issue and added a guide on how to develop the plugin locally: [15363be](https://github.com/ivan-lednev/obsidian-day-planner/commit/15363be851fbfd2ef6b27108e260be5e304486b1)
> 
> When it comes to styling, I try to make it as similar to Obsidian's own styles as possible. If you'd like to customize that (make things look different from the default theme), I think a better approach is to add those styles to settings. You might create a separate issue for that or submit a PR.

Thanks for the development guide. That will help out in the next PR. :)

It looks like your fix solves the issue and does it in a better way than I could (mainly due to my still learning svelte). So I will go ahead and close this PR as it is not redundant. I agree with the philosophy, that sticking as close to Obsidian use of classes is better than not.
